core/domain/state_domain.py

```diff
@@ -1461,6 +1466,17 @@ def validate(self, exp_param_specs_dict, allow_null_interaction):
                     'Found a duplicate content id %s' % solution_content_id)
             content_id_list.append(solution_content_id)
 
+        if not isinstance(self.solicit_answer_details, bool):
+            raise utils.ValidationError(
+                'Expected solicit_answer_details to be a boolean, ' +
+                'received %s' % self.solicit_answer_details)
+        if self.solicit_answer_details:
+            if self.interaction.id in feconf.CANNOT_SOLICIT_ANSWER_DETAILS:
+                raise utils.ValidationError(
+                    'This state cannont have true instance of ' +
```

Comment:
The %s interaction does not support soliciting answer details from learners.

Reply:
Done